### PR TITLE
Adding new environment from override, fixes #103

### LINKED ISSUE #103: Adding new environments via .psh.yaml.dist.override fails
See https://github.com/shopware/development/issues/72

### PR DESCRIPTION
**Pull request by peter-sition**

### CODE REVIEW

**Comment by peter-sition** [comment]
The test for PHP 7.0 is failing, but I can't find the exact reason why?

**Comment by JanPietrzyk** [comment]
Should have been a timing issue with the parallel process execution. I have restarted the build.

File: src/Config/ConfigMerger.php [hunk 1]
```diff
@@ -42,14 +42,24 @@ public function merge(Config $config, Config $override = null): Config
...
+
+        foreach ($foundEnvironments as $name) {
+            if (!isset($override->getEnvironments()[$name])) {
+                $environments[$name] = $config->getEnvironments();
+
+                continue;
+            }
+
+            $overrideEnv = $override->getEnvironments()[$name];
+            $originalConfigEnv = $config->getEnvironments()[$name] ?? new ConfigEnvironment();
```
Comment (JanPietrzyk) [inline src/Config/ConfigMerger.php:56]: I am not entirely sure if an empty object will work here. I will merge this onto the 1.5 branch and test it from there.

**Comment by JanPietrzyk** [comment]
Nice implementation! Thank you for your contribution

**Pull request merged**